test/controllers/save_a_page_test.rb

```diff
@@ -0,0 +1,97 @@
+require "test_helper"
+
+class ContentItemsControllerTest < ActionController::TestCase
+  include GdsApi::TestHelpers::ContentStore
+  include GovukPersonalisation::TestHelpers::Requests
+
+  SAVE_A_PAGE_CSS_SELECTOR = ".app-c-save-this-page".freeze
+
+  test "does not show the save-page button when the feature flag is off" do
+    stub_feature_flag_off
+    set_up_and_visit_content_item("detailed_guide", "detailed_guide")
+
+    assert_not has_save_this_page_button_css_selector
+  end
+
+  test "logged out - shows the add_page_button when the feature flag is on" do
+    stub_feature_flag_on
+    stub_saved_pages_unauthenticated("/guidance/salary-sacrifice-and-the-effects-on-paye")
+    set_up_and_visit_content_item("detailed_guide", "detailed_guide")
+
+    assert has_save_page_button("/guidance/salary-sacrifice-and-the-effects-on-paye")
+  end
+
+  test "logged in - shows the add_page_button when the feature flag is on and page is not saved" do
+    stub_feature_flag_on
+    mock_logged_in_session
+    stub_saved_pages_no_page("/guidance/salary-sacrifice-and-the-effects-on-paye")
+    set_up_and_visit_content_item("detailed_guide", "detailed_guide")
+
+    assert has_save_page_button("/guidance/salary-sacrifice-and-the-effects-on-paye")
+  end
+
+  test "logged in - shows the remove_page_button when the feature flag is on and page is saved" do
+    stub_feature_flag_on
+    mock_logged_in_session
+    stub_saved_pages_has_page("/guidance/salary-sacrifice-and-the-effects-on-paye")
+    set_up_and_visit_content_item("detailed_guide", "detailed_guide")
+
+    assert has_remove_this_page_button("/guidance/salary-sacrifice-and-the-effects-on-paye")
+  end
+
+private
+
+  def stub_feature_flag_on
+    @controller.stubs(:save_this_page_enabled?).returns(true)
+  end
+
+  def stub_feature_flag_off
+    @controller.stubs(:save_this_page_enabled?).returns(false)
+  end
+
+  def stub_saved_pages_unauthenticated(page_path)
+    stub_saved_pages_query(page_path, 401)
+  end
+
+  def stub_saved_pages_query(page_path, status, body = {}, logged_in = nil)
+    account_session_header = logged_in ? { "Govuk-Account-Session" => "placeholder" } : {}
+    stub_request(:get, "https://account-api.test.gov.uk/api/saved-pages/#{CGI.escape(page_path)}")
+    .with(
+      headers: {
+        "Accept" => "application/json",
+        **account_session_header,
+      }.compact,
+    ).to_return(status: status, **body)
+  end
+
+  def stub_saved_pages_has_page(page_path)
+    body = { body: { "saved_page": ["some api content here"] }.to_json }
+    stub_saved_pages_query(page_path, 200, body)
+  end
+
+  def stub_saved_pages_no_page(page_path)
+    stub_saved_pages_query(page_path, 404)
+  end
```

Comment:
Use the test helpers in gds-api-adapters, as if the response format changed the pact tests will start to fail - there's no direct connection between the gds-api-adapters pact tests and the test helpers (unfortunately), but seeing the failing pact tests should prompt someone to look at other places we're making assumptions about the response format.